docs/GLOSSARY.md

```diff
@@ -1,76 +1,84 @@
 # Glossary
 
-Index of terminology used in Randovania. Certain terms sometimes are used only in the GUI, with more technical ones being used in code or CLI.
+Index of terminology used in Randovania. Some terms are used only in the GUI, with more technical ones being used in code or CLI.
 
 ## Area
 
-TODO
+A small chunk of a game's playable space, often only a single room. May contain one or more [Nodes](#node).
+
+Examples:
+- Phazon Mining Tunnel in Metroid Prime
+- Big Pink in Super Metroid
 
 ## Configuration
 
-TODO
+A colloquial term referring to the many types of settings a player may use when generating a game. This includes trick settings, item pool settings, starting items and location, [Gameplay and Cosmetic Patches](#game-patches), and many other things.
 
 ## Dangerous Action
 
-TODO
+Any action that the [Solver](#solver) might take which it might not be able to backtrack from; a one-way action. See also [Point of No Return](#point-of-no-return).
 
 ## Database
 
-TODO
+The collection of data that represents a game's [Logic](#logic), including the layout of [Worlds](#world) and [Areas](#area), the [Resources](#resource) a player has to progress, the locations of [Pickups](#pickup) and spawn points, and other information.
 
 ## Event
 
-TODO
+A logical [Resource](#resource) representing an event that may be triggered in game. Events can only be completed once, and cannot be undone.
 
 ## Game Patches
 
-TODO
+A modification made to the game. Usually refers to specific changes made to the game, such as cutscene alterations, and not to the randomization process itself.
 
 ## Generator
 
-The part of Randovania that takes the [database](#database), the [preset](#preset) settings, a [random number generator](#rng) and decides which [pickups](#pickup) should be placed where, as well as everything else that is randomly decided in the [layout](#layout).
+The part of Randovania that takes the [Database](#database), the [Preset](#preset) settings, a [Random Number Generator](#rng) and decides which [Pickups](#pickup) should be placed where, as well as everything else that is randomly decided in the [Layout](#layout).
 
 ## Hint
 
-TODO
+A piece of text placed in the game world which may give the player information about how [Pickups](#pickup) were placed. This is distinct from [Log Files](#log-file), which contain all details of a generated game, and are generally considered to be a form of cheating in most competitive settings.
 
 ## Item
 
-A kind of [resource](#resource) that represents something a player has during the game.
+A kind of [Resource](#resource) that represents something a player has during the game.
 
 Examples:
 - Space Jump Boots in Metroid Prime 1/2.
 - Missiles in Metroid games.
 
+## Item Pool
```

Comment:
While there is a couple places in randovania that uses this term, it's an incorrect term we shouldn't ever use. The correct is Pickup Pool.

So it's a colloquial term at best, and if used it should be in the gui.